doc/source/cluster/kubernetes/user-guides/kuberay-gcs-ft.md

```diff
@@ -0,0 +1,259 @@
+    ```
+
+### Step 5: Create a detached actor
+
+```sh
+# Step 4.1: Create a detached actor with name `counter_actor`.
+export HEAD_POD=$(kubectl get pods --selector=ray.io/node-type=head -o custom-columns=POD:metadata.name --no-headers)
+kubectl exec -it $HEAD_POD -- python3 /home/ray/samples/detached_actor.py
+
+# Step 4.2: Increment the counter.
+kubectl exec -it $HEAD_POD -- python3 /home/ray/samples/increment_counter.py
+
+# 2023-09-07 16:01:41,925 INFO worker.py:1313 -- Using address 127.0.0.1:6379 set in the environment variable RAY_ADDRESS
+# 2023-09-07 16:01:41,926 INFO worker.py:1431 -- Connecting to existing Ray cluster at address: 10.244.0.17:6379...
+# 2023-09-07 16:01:42,000 INFO worker.py:1612 -- Connected to Ray cluster. View the dashboard at http://10.244.0.17:8265
+# 1
+```
+
+(kuberay-external-storage-namespace-example)=
+### Step 6: Check the data in Redis
+
+```sh
+# Step 6.1: Check the RayCluster's UID.
+kubectl get rayclusters.ray.io raycluster-external-redis -o=jsonpath='{.metadata.uid}'
+# [Example output]: 864b004c-6305-42e3-ac46-adfa8eb6f752
+
+# Step 6.2: Check the head Pod's environment variable `RAY_external_storage_namespace`.
+kubectl get pods $HEAD_POD -o=jsonpath='{.spec.containers[0].env}' | jq
+# [Example output]:
+# [
+#   {
+#     "name": "RAY_external_storage_namespace",
+#     "value": "864b004c-6305-42e3-ac46-adfa8eb6f752"
+#   },
+#   ...
+# ]
+
+# Step 6.3: Log into the Redis Pod.
+export REDIS_POD=$(kubectl get pods --selector=app=redis -o custom-columns=POD:metadata.name --no-headers)
+# The password `5241590000000000` is defined in the `redis-config` ConfigMap.
+kubectl exec -it $REDIS_POD -- redis-cli -a "5241590000000000"
+
+# Step 6.4: Check the keys in Redis.
+KEYS *
+# [Example output]:
+# 1) "864b004c-6305-42e3-ac46-adfa8eb6f752"
+
+# Step 6.5: Check the value of the key.
+HGETALL 864b004c-6305-42e3-ac46-adfa8eb6f752
+```
+
+In [ray-cluster.external-redis.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.external-redis.yaml), the `ray.io/external-storage-namespace` annotation isn't set for the RayCluster.
+Therefore, KubeRay automatically injects the environment variable `RAY_external_storage_namespace` to all Ray Pods managed by the RayCluster with the RayCluster's UID as the external storage namespace by default.
+
+### Step 7: Kill the GCS process in the head Pod
+
+```sh
+# Step 7.1: Check the `RAY_gcs_rpc_server_reconnect_timeout_s` environment variable in both head Pod and worker Pod.
+kubectl get pods $HEAD_POD -o=jsonpath='{.spec.containers[0].env}' | jq
+# [Expected result]:
+# No `RAY_gcs_rpc_server_reconnect_timeout_s` environment variable is set. Hence, Ray head will use its default value `60`.
+kubectl get pods $YOUR_WORKER_POD -o=jsonpath='{.spec.containers[0].env}' | jq
+# [Expected result]:
+# KubeRay injects the `RAY_gcs_rpc_server_reconnect_timeout_s` environment variable with the value `600` to the worker Pod.
```

Comment:
At this point, the reader might wonder why we're checking these values.  Maybe it's worth briefly explaining why the worker pod timeout should be much longer than the head pod timeout

Reply:
Updated https://github.com/ray-project/ray/pull/39450/commits/ef9ce5cf3bfbe794c54d27cfeca41ed56068e364